update mom6 to its main repo. 20250818 commit

jiandewang opened the pull request:
<!-- INSTRUCTIONS: 
- PLEASE READ/FOLLOW THE DIRECTIONS IN EACH SECTION
- Complete the 'Commit Queue Requirements' below
- Please use github markup as much as possible (https://docs.github.com/en/get-started/writing-on-github)
- Please leave your PR in a draft state until all underlying work is completed.
-->
## Commit Queue Requirements:
<!--
- Please complete the items that follow this.
- Please "check off" completed items. Use [X] for a filled in checkbox or leave it [ ] for an empty checkbox
- Your PR will not be considered until all requirements are met.
- THIS IS YOUR RESPONSIBILITY
 -->
- [x] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [ ] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
## Description:
<!--
Please provide a detailed verbose description of what this PR does
-->
MOM6 main repo. is upated on 20250818 which is originally GFDL's change default parameter PR (https://github.com/mom-ocean/MOM6/pull/1670). Need to merge it into dev/emc. See detail at EMC issue https://github.com/NOAA-EMC/MOM6/issues/160

### Commit Message:
<!--
Please provide concise information for The UFS-WM and/or each sub-component:
Please delete what is not needed.
-->
```
  * MOM6 - update MOM6 to its main repo. 20250818 (default parameter changes) and cleanup MOM_input
```
### Priority:
<!--
Please provide the priority you would prefer this pull request to have.
* Critical Bugfix: Model is wrong.
* High: Time-sensitive project.
* Normal.
Please delete the ones that are not applicable
-->
* Normal
## Git Tracking
### UFSWM:
<!--
Please add the UFS-WM github issue here if there is one
Please delete the one that is not applicable.
-->
* Closes #2865 

### Sub component Pull Requests:
<!--
Please provide a list of sub-components involved with this pull request.
Please provide links to the sub-component pull requests as shown below.
Please delete what is not needed.
Example:
* FV3: NOAA-EMC/fv3atm#734
  * ccpp-physics: ufs-community/ccpp-physics#33
* WW3: NOAA-EMC/WW3#321
-->
* MOM6: PR161 https://github.com/NOAA-EMC/MOM6/pull/161)
### UFSWM Blocking Dependencies:
<!--
If there are any UFSWM PR's that are needed to be completed before this one, please add links
to them here
Please delete what is not needed.
-->
* None

### Documentation:
<!--
Indicate what documentation, if any, is needed for this PR and who will add it (if applicable).
Please delete what is not needed.
-->
* No documentation update is required for this PR (please explain).

---
## Changes
### Regression Test Changes (Please commit test_changes.list):
<!--
Please let us know if this PR creates new baselines, changes baselines or not.
Please delete what is not needed.
Please make sure you have properly submitted test_changes.list
-->
* PR Updates/Changes Baselines.

### Input data Changes:
<!--
If there are any changes to input-data for a test, please provide information here.
Please delete what is not needed.
-->
* None.

### Library Changes/Upgrades:
<!-- Library updates take time. Please provide library and version information here.
** SPECIAL INSTRUCTIONS **
If this PR needs updates to libraries please make sure to accomplish the following tasks:
- Create separate issue in (https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
- Add issue link from JCSDA/spack-stack following this item <!-- for example: "* JCSDA/spack-stack#1757"

Please delete what is not needed.
-->
* No Updates
  
---
<!-- STOP!!! THE FOLLOWING IS FOR CODE MANAGERS ONLY. PLEASE DO NOT FILL OUT -->
## Testing Log:
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] GaeaC6
  - [ ] Derecho
  - [ ] Ursa
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

DeniseWorthen commented on tests/parm/MOM_input_025.IN:
```diff
@@ -193,6 +195,11 @@ NK = 75                         !   [nondim]
 ! === module MOM_tracer_registry ===
 
 ! === module MOM_EOS ===
+EQN_OF_STATE = "WRIGHT"         ! default = "WRIGHT_FULL"
```
@jiandewang What is the difference between WRIGHT and WRIGHT_FULL?

jiandewang commented:
@DeniseWorthen I don't know that much in detail on this, this is what original PR said:

The default equation of state is now EQN_OF_STATE = "WRIGHT_FULL", in place of the buggy previous default of "WRIGHT"

DeniseWorthen commented:
So why would we want to use "buggy" WRIGHT EOS?

jiandewang commented:
actually I hesitated a bit last night when I modify that section, let me switch to the new default option

jiandewang commented:
@sanAkel FYI

sanAkel commented:
@jiandewang It makes no sense to use known buggy options!

Cc: @yangfanglin

DeniseWorthen commented:
Looking at the problematic routine, I also see ``WAVE_INTERFACE_ANSWER_DATE`` is available. Have you tried modifying that and leaving ``DEFAULT_ANSWER_DATE`` at the 2025 value?

jiandewang commented:
@DeniseWorthen that is another option in my mind, I am testing 
https://github.com/mom-ocean/MOM6/blob/main/src/user/MOM_wave_interface.F90#L334-L338
trying to narrow down the exact date that causing trouble

jiandewang commented:
test indicates WAVE_INTERFACE_ANSWER_DATE = 20230101 works but 20230102 and after failed.
I am going to use DEFAULT_ANSWER_DATE = latest but explicit set wave to 20230101

jiandewang commented:
the above setting worked for all test cases, will summarize issue and work with GFDL. Hopefully we can remove wave answer date in the next round of PR https://github.com/mom-ocean/MOM6/issues/1676

jiandewang commented:
just removed all "BUG" flags in MOM_input (as we set ENABLE_BUGS_BY_DEFAULT=F now)

jiandewang commented:
@DeniseWorthen can you help me double check MOM_input ? I added ENABLE_BUGS_BY_DEFAULT = False and removed all other BUG flags (not really too much of that, GRID_ROTATION_ANGLE_BUGS and USE_TRIPOLAR_GEOLONB_BUG were set as F which mean they are using the bug-fix code so they are harmless but no need to be in MOM_input)

DeniseWorthen commented:
@jiandewang To be honest, I'm not sure what the setting of ``USE_LA_LI2016`` should be when we are running without waves. I know we are currently using it ``T`` in the tests which do not have waves, but I'm not sure that is the right setting.

Can you check w/ someone on the WAV team?

jiandewang commented:
> @jiandewang To be honest, I'm not sure what the setting of `USE_LA_LI2016` should be when we are running without waves. I know we are currently using it `T` in the tests which do not have waves, but I'm not sure that is the right setting.
> 
> Can you check w/ someone on the WAV team?

@JessicaMeixner-NOAA can you tell us who is the right person on wave team for us to ask this question ?

JessicaMeixner-NOAA commented:
@jiandewang  I'll see if I can look into the USE_LA_LI2016 model setting.

JessicaMeixner-NOAA commented:
Okay I just looked back through emails, and USE_LA_LI2016=True when using w/out the wave model and USE_LA_LI2016=False when using the wave model. 

The USE_LA_LI2016 is a logical to use the Li et al. 2016 (submitted) formula to determine the Langmuir number.  When you are coupling w/a wave component, the wave model is helping calculate that instead. 

@jiandewang  - is that what you needed or something else?

jiandewang commented:
@JessicaMeixner-NOAA this means we are using the correct setting for no wave case. Thanks

DeniseWorthen commented:
I just checked a ``cpld_control_pdlib_p8_intel`` run directory I have. It does use waves but has ``USE_LA_LI2016 = True``. It seems from what Jessica wrote it should be ``False`` (since the WAV model is calculating it instead?). 

EDIT: I do see old emails(~2020) w/ a fix of a halo-decomp issue and Alper finding that setting this parameter = F gave restart reproducibility. I think that was a temporary fix, once the halo issue was resolved, we were able to set it ``T`` and still get restart-repro w/o waves.

jiandewang commented:
I just checked all rt.sh cpld cases, they are all T
URSA: /scratch4/NCEPDEV/stmp/Jiande.Wang/RT_RUNDIRS/Jiande.Wang/FV3_RT/rt_885411
grep 'USE_LA_LI2016' cpld*/INPUT/MOM_input

JessicaMeixner-NOAA commented:
I think it's a place where it might be "cleaner" to have it set as false, but it is not expected to change answers when you have a wave component.

jiandewang commented:
just had a quick test use cpld_control_noaero_p8_intel as template,
set USE_LA_LI2016 T or F will give different results

jiandewang commented:
I carefully read Brandon's original email twice. Set USE_LA_LI2016 to T is correct for both with wave and w/o wave.

jiandewang commented:
MOM6 PR has been approved, so this PR can be put in commit queue now. Note it requires a new baseline

sanAkel commented:
> MOM6 PR has been approved, so this PR can be put in commit queue now. Note it requires a new baseline
> 
> 

May I please get a status update and/or ETA when any action to merge this PR will be taken?

My understanding is that all tests have ✅ and baseline answers change for reasons we understand.

cc: @yangfanglin

FernandoAndrade-NOAA commented:
This PR is next for processing, please resync and include an update to test_changes.list to confirm tests that will be affected by changes, thank you.

BrianCurtis-NOAA commented:
@FernandoAndrade-NOAA If there is a priority PR, we should focus on that one.

jiandewang commented:
just sync-ed my branch. Will update test_changes.list if no high prioroty PR ahead of mine

jiandewang commented:
updated test_changes.list

FernandoAndrade-NOAA commented:
> @FernandoAndrade-NOAA If there is a priority PR, we should focus on that one.

Apologies, yes #2872 is marked as priority, we will shift over to processing that PR.

jiandewang commented:
just sync-ed the branch again

jiandewang commented:
@FernandoAndrade-NOAA and @jkbk2004 can you run this PR today ? I am assigned to do some sensitity test of v17 which requires this PR to be done in order to be able to use the latest MOM6 code and settings. Thanks

jiandewang commented:
I just changed the priority to high for this PR

JessicaMeixner-NOAA commented:
> @FernandoAndrade-NOAA and @jkbk2004 can you run this PR today ? I am assigned to do some sensitity test of v17 which requires this PR to be done in order to be able to use the latest MOM6 code and settings. Thanks

@jiandewang  use a branch for this testing, please do not wait for this PR to be merged to move forward on that. We are testing many code in branches for GFSv17 because the queue is too slow otherwise.   It was my understanding from @NeilBarton-NOAA  that we didn't need any code changes for the sensitivity testing - so this is also confusing.

jiandewang commented:
@JessicaMeixner-NOAA I had discussion with Neil and Santha on Tuesday and we want to use the bug-free of MOM6 setting and new equation of state. According to commit queue, it is this PR's trun today

JessicaMeixner-NOAA commented:
Either way, this PR being merged should not prevent you from moving forward on testing things for GFSv17.  If you have something that needs to be high priority for GFSv17 - let's discuss offline to coordinate things as it seems like lots has changed since Monday.

BrianCurtis-NOAA commented:
@jiandewang Pull requests are for merging in code that has been fully tested prior to getting a slot in the commit queue. If you are using it to test other things, that's not what the priority queue is meant for.

On another note @jkbk2004 when there are no priority PR's, the commit queue should be organized such that the oldest PR's get priority. If there are older PR's in an "Open" state and are not ready to be in the commit queue, then I would recommend moving it to Draft and having those PR authors move it to Open once the issues are addressed. If there are draft PR's that have been sitting around for a long time (months), then I would recommend closing them and encouraging the PR authors to resubmit when the work is closer to completion.

On that note, there are many "Open" PR's older than this one. Why is this one getting priority over all the others at this point?

Please adjust your teams daily tasking to incorporate checking all PR's on a daily basis and adjusting the commit queue accordingly.

jiandewang commented:
@BrianCurtis-NOAA this PR has been fully tested two weeks ago so it's in mature stage.

sanAkel commented:
> On that note, there are many "Open" PR's older than this one. Why is this one getting priority over all the others at this point?
> 

[Same question as ⬆️ ](https://github.com/ufs-community/ufs-weather-model/pull/2866#issuecomment-3260164418) Is there an ETA for merge regardless of _priority_?

DeniseWorthen commented:
@sanAkel Please see https://github.com/ufs-community/ufs-weather-model/wiki/Commit-Queue

sanAkel commented:
> @sanAkel Please see https://github.com/ufs-community/ufs-weather-model/wiki/Commit-Queue

There is no column with header `ETA for merge`.

DeniseWorthen commented:
That would be the "testing on" date basically. Actual merge usually happens sometime the *next* day after all the platforms have completed.

sanAkel commented:
> That would be the "testing on" date basically. Actual merge usually happens sometime the _next_ day after all the platforms have completed.

<img width="942" height="58" alt="image" src="https://github.com/user-attachments/assets/4b92b02e-bc0d-4a44-9c21-f58dd463441c" />

Says `09/04` and today's date is `09/11`, [please](https://github.com/ufs-community/ufs-weather-model/pull/2866#issuecomment-3282058899)!

FernandoAndrade-NOAA commented:
> > That would be the "testing on" date basically. Actual merge usually happens sometime the _next_ day after all the platforms have completed.
> 
> <img alt="image" width="942" height="58" src="https://private-user-images.githubusercontent.com/18061653/488483270-4b92b02e-bc0d-4a44-9c21-f58dd463441c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc2MjA5MjYsIm5iZiI6MTc1NzYyMDYyNiwicGF0aCI6Ii8xODA2MTY1My80ODg0ODMyNzAtNGI5MmIwMmUtYmMwZC00YTQ0LTljMjEtZjU4ZGQ0NjM0NDFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTExVDE5NTcwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkMmM3YWMxYzRiNzQ5NGFkMTdkMWJhMzgxNmQxNDU1YzYwYmVkZDExNGRlMWNhMWY1NGM0MGYwZDJkYjYwMjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4MJ62dHfZZqd_Ge7Q0-J0be-kDZmTKIuBLIl7187zx0">
> Says `09/04` and today's date is `09/11`, [please](https://github.com/ufs-community/ufs-weather-model/pull/2866#issuecomment-3282058899)!

Page updated. Merging typically carries into the day after testing begins provided no issues are encountered during testing.

FernandoAndrade-NOAA commented:
@jiandewang I'm seeing comparison failures in `cpld_control_gfsv17_iau_intel` and `atm_ds2s_docn_dice_intel` tests on Ursa, please confirm if these tests are expected to be affected by the changes as well.

jiandewang commented:
For the cpld job it is supposed to change answer. Sorry I missed that in the file list. I will check the second one once I am in front of computer. Thanks

jiandewang commented:
@FernandoAndrade-NOAA yes atm_ds2s_docn_dice_intel also needs a new baseline.
I just update the test_changes.list

DeniseWorthen commented:
I think I got caught w/ one of these same failures; the ``atm_ds2s_docn_dice`` compares to ``cpld_control_nowave_noaero_p8``; so if that test changes, I believe the docn_dice test will also change. 

I'm not sure an issue was created, but I think we need to fix this, it is just too confusing @NickSzapiro-NOAA

NickSzapiro-NOAA commented:
I'm not sure I understand what happened here with the test_changes.list. Did it come from rt.sh or was it made "by hand"?

jiandewang commented:
@NickSzapiro-NOAA when I tested this PR, I deleted all atmos. only, HAFS and all those jobs that ocean is not involved. But somehow I mistakely deleted two jobs that I shouldn't.

gspetro-NOAA commented:
All tests eventually passed on Orion and Hercules, although I had to rerun the missed tests that Fernando mentioned above and append the results to the original log, which showed those three as failures.

gspetro-NOAA commented on tests/test_changes.list:
```diff
@@ -1,98 +1,63 @@
+datm_cdeps_mx025_gefs intel
+datm_cdeps_stochy_gefs intel
+hafs_regional_storm_following_1nest_atm_ocn_wav_mom6 intel
+cpld_control_gfsv17_iau_intel
```
@FernandoAndrade-NOAA Shouldn't `cpld_restart_gfsv17_iau_intel` also be included here?
```suggestion
cpld_control_gfsv17_iau_intel
cpld_restart_gfsv17_iau_intel
```

jiandewang replied:
> @FernandoAndrade-NOAA Shouldn't `cpld_restart_gfsv17_iau_intel` also be included here?

just added it.

DeniseWorthen replied:
Why would a restart test be added to the test change list? When comparing against an existing baseline, which should be what generates the ``test_changes.list``, the restart test will not run if the control test did not reproduce. The restart test is a dependent test


```
RUN | cpld_restart_gfsv17                               | - noaacloud                          |          | cpld_control_gfsv17
RUN | cpld_restart_gfsv17_iau                           | - noaacloud                          |          | cpld_control_gfsv17_iau
```

gspetro-NOAA replied:
@jiandewang Apologies-it appears I was mistaken. Could you remove the restart test? As @DeniseWorthen pointed out, only tests that generate baselines will appear in that file.

jiandewang replied:
> @jiandewang Apologies-it appears I was mistaken. Could you remove the restart test? As @DeniseWorthen pointed out, only tests that generate baselines will appear in that file.

removed. Next time I should ran the whole rt.sh as @dpsarmie mentioned regardless of whatever model is being involved in the PR

gspetro-NOAA replied:
@jiandewang Thanks!

dpsarmie commented:
Added https://github.com/ufs-community/ufs-weather-model/issues/2855 to the description since this PR will also close that issue.

dpsarmie commented on tests/test_changes.list:
```diff
@@ -60,4 +60,5 @@ datm_cdeps_mx025_gefs intel
 datm_cdeps_stochy_gefs intel
 hafs_regional_storm_following_1nest_atm_ocn_wav_mom6 intel
 cpld_control_gfsv17_iau_intel
```
There shouldn't be any underscores between the test name and the compiler.

dpsarmie replied:
@jkbk2004 @jiandewang This still needs to be changed. `cpld_control_gfsv17_iau_intel` and `atm_ds2s_docn_dice_intel` have an extra underscore.

jiandewang replied:
thanks for your careful checking, done

dpsarmie commented:
I know that there are a lot of regression tests and that it takes a while to do so, but maybe we should be enforcing running the full suite to avoid these issues in future PRs? There are a lot of formatting and dependency issues that `rt.sh` needs that could be missed otherwise.

gspetro-NOAA commented:
@dpsarmie I think this would make sense when the failure is due to an incomplete `test_changes.list` file being added to be sure we don't miss anything. I'm not sure it's the best use of resources in something like a case timeout failure, but I admittedly have less experience in all of this than you all. It would be good to discuss at the CM meeting today. I will say that just rerunning these three tests and their dependencies on Hercules took ~1.25 hours as opposed to ~1.75 hours for the full RT suite, so time wise, we're not saving that much by rerunning the failed tests only, although we'd save on compute allocation.

DeniseWorthen commented:
@gspetro-NOAA The more important issue, imho, is that PRs need to be vetted before going to the top of the Q. For example, someone says 'oh, I ran on wcoss2 or gaeac6 and all is good'. Well, there are no GNU tests on either of those two platforms. That means that when testing starts, time is wasted because we don't know the test_change list is actually accurate and, as you point out, things need to be re-run if it turns out it isn't.

This is not the first time that tests are being added to the test_change list after testing has begun. This is a real issue if we ever need to go back and determine when or why a test changed. We need clean testing, clean logs and clean rules for putting things at the top of the Q.

gspetro-NOAA commented:
@DeniseWorthen Agreed!

FernandoAndrade-NOAA commented:
Testing is complete, leaving a note in sub PRs to continue with merge.

jiandewang commented:
Out for lunch at this moment, will be in front of computer in 20 minutes

jiandewang commented:
MOM6 merged with hash fe9e7bf
UFS submodule reverted

jiandewang commented:
sorry I reverted submodule before I saw the approval

DeniseWorthen commented:
@jiandewang @JessicaMeixner-NOAA I ran the ``datm_cdeps_mx025_gefs_intel`` test for 2 days with this PR and the hash previously. This is the SST difference from ``ocn_2011_10_02_21.nc``. It doesn't look like there is anything dramatic changing---the biggest change seems to be a very small cooling in what I would guess are turbulent regions in the Southern Ocean.


<img width="800" height="553" alt="Screenshot 2025-09-16 at 12 40 10 PM" src="https://github.com/user-attachments/assets/8ec642ed-127b-4b9c-897a-fb032be3c58b" />

JessicaMeixner-NOAA commented:
Thank you @DeniseWorthen !!!

jiandewang commented:
@DeniseWorthen Thanks

jiandewang commented:
FYI @sanAkel @NeilBarton-NOAA

sanAkel commented:
> FYI @sanAkel @NeilBarton-NOAA 

That is/was the expectation - I.e., these bug fixes do not change the answers in any "systematic way".

NeilBarton-NOAA commented:
Thanks @DeniseWorthen !
